llvm/lib/CAS/CASDB.cpp

```diff
@@ -18,17 +19,42 @@ void CASDB::anchor() {}
 
 LLVM_DUMP_METHOD void CASID::dump() const { print(dbgs()); }
 LLVM_DUMP_METHOD void CASDB::dump() const { print(dbgs()); }
+LLVM_DUMP_METHOD void ReferenceBase::dump() const { print(dbgs()); }
 
 std::string CASID::toString() const {
   std::string S;
   raw_string_ostream(S) << *this;
   return S;
 }
 
-/// Default implementation opens the file and calls \a createBlob().
+void ReferenceBase::print(raw_ostream &OS) const {
+#if LLVM_ENABLE_ABI_BREAKING_CHECKS
+  if (CAS) {
+    Optional<CASID> ID;
+    if (IsHandle)
+      ID = CAS->getObjectID(ObjectHandle(*CAS, InternalRef));
+    else
+      ID = CAS->getObjectID(ObjectRef(*CAS, InternalRef));
+    OS << *ID;
+    return;
```

Comment:
I would suggest printing the internal-ref as well so that you get a superset of the information.

Reply:
Reworked this substantially in the just-pushed update f766105ab52ea8cd9dd828ce2b65c49b219305b9, taking your suggestion.